File: agent-driven-architecture-main/examples/typescript/src/blocks/triage/slice.ts
// ── blocks/triage/slice — the block's own state, copy-on-write ─────────────
// PURE. No I/O, no clock, no Actor. Every transition returns a NEW slice; a
// transition that cannot be applied returns nothing, so the ARM has to decide
// what to fold instead (12.4 rule 1).

import type { TicketId } from "@adr/spine/pure/ids";
import type { Priority } from "./contract";

export interface Ticket {
  readonly id: TicketId;
  readonly body: string;
}

export interface TriageSlice {
  readonly tickets: ReadonlyMap<TicketId, Ticket>;
  readonly priority: ReadonlyMap<TicketId, Priority>;
}

export const emptyTriageSlice: TriageSlice = { tickets: new Map(), priority: new Map() };

export function triageSliceOf(tickets: readonly Ticket[], level: Priority = "Normal"): TriageSlice {
  return {
    tickets: new Map(tickets.map((t) => [t.id, t])),
    priority: new Map(tickets.map((t) => [t.id, level])),
  };
}

export function ticketOf(slice: TriageSlice, ticket: TicketId): Ticket | null {
  return slice.tickets.get(ticket) ?? null;
}

export function priorityOf(slice: TriageSlice, ticket: TicketId): Priority | null {
  return slice.priority.get(ticket) ?? null;
}

export function withPriority(slice: TriageSlice, ticket: TicketId, level: Priority): TriageSlice {
  const priority = new Map(slice.priority);
  priority.set(ticket, level);
  return { ...slice, priority };
}

File: agent-driven-architecture-main/examples/kotlin/block/triage/src/main/kotlin/adr/blocks/triage/Slice.kt
// ── blocks/triage/slice — the block's own state, and its pure transitions ──
// Copy-on-write; never mutate the input. Structural equality is what makes the
// replay comparison and the golden-effect diff meaningful (14.1).

package adr.blocks.triage

import adr.contract.TriageResult.Priority
import adr.spine.pure.TicketId

// `Priority` is declared on this block's own sealed transport root, in
// blocks/triage/Contract.kt, because Kotlin's sealed rule authors that file inside
// `:spine` and the transport NAMES this type (ADR-001 §3). The import is what C2's
// name-prefix rule admits: `TriageResult.Priority` starts with `Triage`.

data class Ticket(val id: TicketId, val body: String)

data class TriageSlice(
    val tickets: Map<TicketId, Ticket> = emptyMap(),
    val priority: Map<TicketId, Priority> = emptyMap(),
) {
    // No companion: a companion member has no instance, which is the same defect as a
    // top-level function. The EMPTY slice is now what the primary constructor builds
    // when told nothing — `TriageSlice()` — so the shape carries its own starting value and
    // nothing extra has to exist to hand it over.
    fun withPriority(ticket: TicketId, level: Priority): TriageSlice =
        copy(priority = priority + (ticket to level))
}
